### PR TITLE
Listen to fedora messaging

### PR DESCRIPTION
**Pull request by rpitonak**
TODO:

- [x] add queue_name and routing_keys to service config
- [x] merge PR in packit https://github.com/packit-service/packit/pull/548
- [x] merge PR in deployment https://github.com/packit-service/deployment/pull/17
- [x] merge [new secrets](https://gitlab.cee.redhat.com/user-cont/packit-service/merge_requests/45)
- [x] the CI is green
- [x] test everything together on stg (https://github.com/packit-service/hello-world/pull/24)

### CODE REVIEW

**Comment by jpopelka** [comment]
Is this supposed to completely replace `fedmsg`?
If yes, there are more `fedmsg` bits, like
- https://github.com/packit-service/packit-service/blob/master/setup.cfg#L39
- https://github.com/packit-service/packit-service/blob/master/files/fedmsg-ssl.py
- https://github.com/packit-service/deployment/blob/master/openshift/deployment-fedmsg.yml.j2#L61

**Comment by rpitonak** [comment]
> * https://github.com/packit-service/deployment/blob/master/openshift/deployment-fedmsg.yml.j2#L61

Thank you, I am aware of those. This is still work in progress and few changes will follow-up also in deployment repository.

File: packit_service/fedmsg/consumer.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    @staticmethod
+    def fedora_messaging_callback(message: Message):
+        print(f"{message.topic}: {message.body}")
+
+    @staticmethod
+    def consume_from_fedora_messaging():
+        """
+        fedora-messaging is written in an async way: callbacks
+        """
+        queue_name = "708D1D74-63E4-472A-88E8-8E43C5AE40DC"
```
Comment (TomasTomecek) [inline packit_service/fedmsg/consumer.py]: huh, what?

File: packit_service/fedmsg/consumer.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                "auto_delete": True,  # Delete the queue when the client terminates
+                "exclusive": False,  # Allow multiple simultaneous consumers
+                "arguments": {},
+            }
+        }
+        binding = {
+            "exchange": "amq.topic",  # The AMQP exchange to bind our queue to
+            "queue": queue_name,  # The unique name of our queue on the AMQP broker
+            # The topics that should be delivered to the queue
+            "routing_keys": ["org.fedoraproject.prod.copr.build.#"],
```
Comment (TomasTomecek) [inline packit_service/fedmsg/consumer.py]: can we either make this an argument of the method or rename the method?

File: packit_service/fedmsg/consumer.py (hunk 1 above)
Comment (rpitonak) [reply packit_service/fedmsg/consumer.py]: from [docs](https://fedora-messaging.readthedocs.io/en/latest/quick-start.html#fedora-s-public-broker):

```
# Queue names *must* be in the normal UUID format: run "uuidgen" and use the
# output as your queue name. If your queue is not exclusive, anyone can connect
# and consume from it, causing you to miss messages, so do not share your queue
# name. Any queues that are not auto-deleted on disconnect are garbage-collected
# after approximately one hour.
```

File: packit_service/fedmsg/consumer.py (hunk 2 above)
Comment (rpitonak) [reply packit_service/fedmsg/consumer.py]: sure, this was just to try it out

File: packit_service/fedmsg/consumer.py (hunk 1 above)
Comment (jpopelka) [reply packit_service/fedmsg/consumer.py]: Please add a comment about who generated the UUID.

> do not share your queue name

Do we care? Shouldn't it be a secret then?

Comment (TomasTomecek) [reply packit_service/fedmsg/consumer.py]: secret +1

File: packit_service/worker/fedmsg_handlers.py [hunk 3]
```diff
@@ -132,3 +140,85 @@ def run(self) -> HandlerResults:
...
+        r = BuildStatusReporter(self.event.get_project(), build["commit_sha"])
+        url = (
+            "https://copr.fedorainfracloud.org/coprs/"
+            f"{self.event.owner}/{self.event.project_name}/build/{self.event.build_id}/"
+        )
+
+        msg = "RPMs failed to be built."
+        gh_state = "failure"
+
+        if self.event.status == 1:
```
Comment (TomasTomecek) [inline packit_service/worker/fedmsg_handlers.py]: `status == 1`, this look awful, please at least comment what this means

File: packit_service/worker/jobs.py [hunk 4]
```diff
@@ -83,6 +85,9 @@ def process_jobs(self, event: Event) -> Dict[str, HandlerResults]:
         """
         Run a job handler (if trigger matches) for every job defined in config.
         """
+
+        logger.warning("PROCESS JOBS")
```
Comment (TomasTomecek) [inline packit_service/worker/jobs.py]: !!!!!!

Comment (TomasTomecek) [reply packit_service/worker/jobs.py]: (this was a joke, to be honest :D I just found it amusing the log message was written in all caps)

**Comment by rpitonak** [comment]
recheck

**Comment by rpitonak** [comment]
recheck

**Comment by rpitonak** [comment]
recheck

**Comment by rpitonak** [comment]
recheck

**Comment by rpitonak** [comment]
recheck

**Comment by rpitonak** [comment]
Verified on stg also. (The problem is just that hello-world build failed in copr)

**Comment by TomasTomecek** [comment]
prod deploy then?

File: packit_service/fedmsg/consumer.py (hunk 1 above)
Comment (jpopelka) [reply packit_service/fedmsg/consumer.py]: @rpitonak ok, so it's in secrets now, great. But I still don't have an answer to my question about what/where/how was the UUID generated? Or where can I see this queue? Or basically, where should I go when something goes wrong?

Comment (jpopelka) [reply packit_service/fedmsg/consumer.py]: https://gitlab.cee.redhat.com/user-cont/packit-service/issues/8

**Pull request merged**